### PR TITLE
Create Wordle-like game service

### LINKED ISSUE #3352: Wordle service
It might be nice to have a Wordle service which facilitates a game of Wordle like the Tic Tac Toe game.

### PR DESCRIPTION
**Pull request by gsteinLTU**
Close #3352 

Target words are drawn from CommonWords, allowed guesses are from a custom dictionary.

Screenshot from 2022-01-27 17-03-29
[Example project in ZIP](https://github.com/NetsBlox/NetsBlox/files/7954552/Wordle.zip)

### CODE REVIEW

**Comment by gsteinLTU** [comment]
Screenshot from 2022-01-27 18-16-14
[Example guessing project in ZIP](https://github.com/NetsBlox/NetsBlox/files/7954685/Wordle.Guesser.zip)

**Comment by gsteinLTU** [comment]
@brollb Let me know if there's a different way you'd prefer this to work. I looked at the older game services, but they didn't seem to be handling separate states for separate rooms when I tested them (which I thought they used to).

File: src/server/services/procedures/wordguess/wordguess.js [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+        } else {
+            matches.push(1);
+        }
+    }
+
+    logger.log(realWord);
+
+    // Find near-match
+    for (let i = 0; i < realWord.length; i++) {
+        if (realWord.indexOf(word[i]) != -1) {
```
Comment (brollb) [inline src/server/services/procedures/wordguess/wordguess.js:138]: `.includes` is a little syntactic sugar for this

**Review by brollb — CHANGES_REQUESTED** [review]
Nice work! Currently, services that maintain separate state for each room are implemented as a constructor rather than an object and store their state under `_state`. I think that should be sufficient for this service. Here is an example of a service written to be instantiated for each requesting project: https://github.com/NetsBlox/NetsBlox/blob/master/src/server/services/procedures/simple-hangman/simple-hangman.js#L30-L37

The only pressing change is marking it as an alpha service so it isn't available on editor until it is ready. A couple other changes that would be good:
- `serviceName` can be removed if you rename the directory (& file) to `word-guess`
- Refactor the service to follow the pattern from simple-hangman so it doesn't have to manage cleaning up states, etc. However, if you want to scope it based on client ID rather than project, you could store the state in the database and set a ttl index.

File: src/server/services/procedures/wordguess/wordguess.js (hunk 1 above)
Comment (gsteinLTU) [reply src/server/services/procedures/wordguess/wordguess.js]: Done

**Comment by gsteinLTU** [comment]
@brollb Renamed the service. I was aware of `_state` already, but it wasn't working how I remembered it working when I tried using it. I opened an issue about that, since you're describing it working the way I remembered: #3358

**Comment by brollb** [comment]
Sounds good. We can merge it as-is until we check out what is going on with #3358.

Thanks, @gsteinLTU!

**Pull request merged**